assets/js/blocks/cart-checkout/checkout/terms-and-conditions-checkbox/index.tsx

```diff
@@ -0,0 +1,136 @@
+/**
+ * External dependencies
+ */
+import { useState, ReactElement, useEffect } from 'react';
+import CheckboxControl from '@woocommerce/base-components/checkbox-control';
+import { __ } from '@wordpress/i18n';
+import { useCheckoutContext } from '@woocommerce/base-context';
+import { FormStep } from '@woocommerce/base-components/cart-checkout';
+import { TERMS_URL, TERMS_PAGE_NAME } from '@woocommerce/block-settings';
+import { createInterpolateElement } from 'wordpress-element';
+import { getSetting } from '@woocommerce/settings';
+
+/**
+ * Internal dependencies
+ */
+import { useValidationContext } from '../../../../base/context/shared';
+import blockAttributes from '../attributes';
+
+interface TermsAndConditionsCheckboxProps {
+	text: string;
+}
+
+export const TermsAndConditionsCheckbox = ( {
+	text,
+}: TermsAndConditionsCheckboxProps ): ReactElement => {
+	const {
+		isProcessing: checkoutIsProcessing,
+		isTermsAcceptedCheckboxPristine,
+		dispatchActions,
+	} = useCheckoutContext();
+
+	const {
+		setValidationErrors,
+		clearValidationError,
+		getValidationError,
+	} = useValidationContext();
+
+	const defaultTermsAndConditionsText =
+		blockAttributes.termsAndConditionsText.default;
+
+	const [ termsAndConditionsText, setTermsAndConditionsText ] = useState<
+		string
+	>( defaultTermsAndConditionsText );
+
+	useEffect( () => {
+		let termsText = text?.trim();
+		if ( ! termsText ) {
+			termsText = defaultTermsAndConditionsText;
+		}
+		if (
+			typeof termsText === 'string' &&
+			termsText.includes( '[terms]' )
+		) {
+			termsText = termsText.replace( /\[terms\]/, '<terms/>' );
+		}
+		setTermsAndConditionsText( termsText );
+	}, [ getSetting, text ] );
+
+	const isCheckboxInvalid =
+		! isTermsAcceptedCheckboxPristine &&
+		getValidationError( 'terms_and_conditions' );
+
+	const setValidationError = () => {
+		setValidationErrors( {
+			terms_and_conditions: {
+				message: __(
+					'You must agree to the terms and conditions to continue.',
```

Comment:
In the current PR, the terms text can be changed into anything. Does that affect this notice? Should we inherit anything from the terms page title?